gui/wxpython/datacatalog/tree.py

```diff
@@ -854,22 +862,112 @@ def OnRenameLocation(self, event):
+            self.showNotification.emit(message=label)
+            event.Veto()
+        else:
+            if node.data['type'] == 'mapset':
+                mapset = self.selected_mapset[0].data['name']
+                message = get_reason_mapset_not_removable(self.selected_grassdb[0].data['name'],
+                                                          self.selected_location[0].data['name'],
+                                                          mapset,
+                                                          check_permanent=True)
+                if message:
+                    label = _(
+                        "Cannot rename mapset <{mapset}> for the "
+                        "following reason: {reason}").format(
+                        mapset=mapset, reason=message)
+                    self.showNotification.emit(message=label)
+                    event.Veto()
+                    return
+            elif node.data['type'] == 'location':
+                location = self.selected_location[0].data['name']
+                messages = get_reasons_location_not_removable(self.selected_grassdb[0].data['name'],
+                                                              location)
+                if messages:
+                    label = _(
+                        "Cannot rename location <{location}> for the "
+                        "following reasons: {reasons}").format(
+                        location=location, reasons="\n".join(messages))
+                    self.showNotification.emit(message=label)
+                    event.Veto()
+                    return
+
+            Debug.msg(1, "Start label edit {name}".format(name=node.data['name']))
+            label = _("Editing {name}").format(name=node.data['name'])
+            self.showNotification.emit(message=label)
 
     def OnEditLabel(self, node, event):
         """End label editing"""
-        if self.selected_layer and not event.IsEditCancelled():
-            old_name = node.data['name']
-            Debug.msg(1, "End label edit {name}".format(name=old_name))
-            new_name = event.GetLabel()
+        if event.IsEditCancelled():
+            return
+
+        old_name = node.data['name']
+        Debug.msg(1, "End label edit {name}".format(name=old_name))
+        new_name = event.GetLabel()
+
+        if node.data['type'] in ('raster', 'raster_3d', 'vector'):
             self.Rename(old_name, new_name)
 
+        elif node.data['type'] == 'mapset':
+            message = check_mapset_name(self.selected_grassdb[0].data['name'],
+                                        self.selected_location[0].data['name'],
+                                        new_name)
+            if message:
+                label = _(
+                    "Cannot rename mapset <{mapset}> for the "
+                    "following reason: {reason}").format(
+                    mapset=old_name, reason=message)
+                self.showNotification.emit(message=label)
```

Comment:
Here, use GError (core/gcmd.py) instead of the notification (user showTraceback=False).